internal/options/alias.go

```diff
@@ -0,0 +1,246 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to execute alias command: %w", err)
+		}
+		err = json.Unmarshal(stdout, &ret)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshal json output of alias command: %w", err)
+		}
+	}
+
+	return ret, nil
+}
+
+const (
+	Literal AliasType = "literal"
+
+	CamelCase      AliasType = "camelCase"
+	PascalCase     AliasType = "pascalCase"
+	SnakeCase      AliasType = "snakeCase"
+	UpperSnakeCase AliasType = "upperSnakeCase"
+	KebabCase      AliasType = "kebabCase"
+	DotCase        AliasType = "dotCase"
+
+	FilePattern AliasType = "filePattern"
+
+	Command AliasType = "command"
+)
+
+// Alias is a catch-all type for alias configurations
+// TODO: can we use OOP?
+type Alias struct {
+	Type AliasType `yaml:"type"`
+
+	// Literal
+	Flags map[string][]string `yaml:"flags,omitempty"`
+
+	// FilePattern
+	Path         *string `yaml:"path,omitempty"`
+	Pattern      *string `yaml:"pattern,omitempty"`
+	FileContents []byte  `yaml:"-"` // data for pattern matching
+
+	// Command
+	Command *string `yaml:"command,omitempty"`
+	Timeout *int64  `yaml:"timeout,omitempty"`
+}
+
+func (a *Alias) IsValid() error {
+	err := a.Type.IsValid()
+	if err != nil {
+		return err
+	}
+
+	// Validate expected fields
+	switch a.Type {
+	case Literal:
+		if a.Flags == nil {
+			return errors.New("literal aliases must provide an 'flags'")
+		}
+	case FilePattern:
+		if a.Path == nil {
+			return errors.New("filePattern aliases must provide a 'path'")
+		}
+		if a.Pattern == nil {
+			return errors.New("filePattern aliases must provide a 'pattern'")
+		}
+		if !strings.Contains(*a.Pattern, "FLAG_KEY") {
+			return errors.New("pattern must contain 'FLAG_KEY' for templating")
+		}
+		_, err := regexp.Compile(*a.Pattern)
+		if err != nil {
+			return fmt.Errorf("could not validate regex pattern: %v", err)
+		}
+	case Command:
+		if a.Command == nil {
+			return errors.New("command aliases must provide a 'command'")
+		}
+		if a.Timeout != nil && *a.Timeout < 0 {
+			return errors.New("field 'timeout' must be >= 0")
+		}
+	}
+
+	// Validate unexpected fields
+	var unexpectedField string
+	switch {
+	case a.Type != Literal:
+		if a.Flags != nil {
+			unexpectedField = "flags"
+		}
+	case a.Type != FilePattern:
+		if a.Path != nil {
+			unexpectedField = "path"
+		}
+		if a.Pattern != nil {
+			unexpectedField = "pattern"
+		}
+	case a.Type != Command:
+		if a.Command != nil {
+			unexpectedField = "command"
+		}
+		if a.Timeout != nil {
+			unexpectedField = "timeout"
+		}
+	}
+	if unexpectedField != "" {
+		return a.Type.unexpectedFieldErr(unexpectedField)
+	}
+
+	return nil
+}
+
+type YamlOptions struct {
+	Aliases []Alias `yaml:"aliases"`
+}
+
+func (o *YamlOptions) IsValid() error {
+	for _, a := range o.Aliases {
+		err := a.IsValid()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Yaml() (*YamlOptions, error) {
+	pathToYaml := filepath.Join(Dir.Value(), ".launchdarkly/config.yaml")
```

Comment:
Support just `.yaml` for now - viper should add support for `.yml` automatically later